pinot-broker/src/main/java/com/linkedin/pinot/broker/api/resources/PinotBrokerTimeBoundaryDebug.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.broker.api.resources;
+
+import com.linkedin.pinot.broker.routing.TimeBoundaryService;
+import com.linkedin.pinot.common.config.TableNameBuilder;
+import com.linkedin.pinot.common.utils.CommonConstants;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/*
+ * A Rest API to print the time boundary info for a table, if it exists.
+ * Examples:
+ *   curl broker:port/debug/timeBoundary/table will print the time boundary of offline table.
+ *   curl broker:port/debug/timeBoundary/table_OFFLINE will do the same as above.
+ *
+ * Realtime tables do not have time boundary as of now, so it will return an empty json object
+ *
+ */
+@Api(tags = "Debug")
+@Path("/debug")
+public class PinotBrokerTimeBoundaryDebug {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PinotBrokerTimeBoundaryDebug.class);
+
+  @Inject
+  private TimeBoundaryService timeBoundaryService;
+
+  @GET
```

Comment:
FYI, this method can be part of the same class above since it shares some annotations

(not a required change)

Reply:
combined